llvm/lib/Target/AIE/AIEBaseSubtarget.cpp

```diff
@@ -446,17 +446,39 @@ class WAWEdges : public ScheduleDAGMutation {
       }
     }
   }
-  void apply(ScheduleDAGInstrs *DAG) override {
+
+public:
+  void apply(ScheduleDAGInstrs *DAG, AIEPostRASchedStrategy *Scheduler) {
```

Comment:
Maybe it is better to not pass Scheduler as parameter here because hidden overloads causing warnings.

One idea to avoid this warning:

```
--- a/llvm/lib/Target/AIE/AIEBaseSubtarget.cpp
+++ b/llvm/lib/Target/AIE/AIEBaseSubtarget.cpp
@@ -447,8 +447,15 @@ class WAWEdges : public ScheduleDAGMutation {
     }
   }
 
+  AIEPostRASchedStrategy *Scheduler = nullptr;
+
 public:
-  void apply(ScheduleDAGInstrs *DAG, AIEPostRASchedStrategy *Scheduler) {
+
+  void setScheduler(AIEPostRASchedStrategy *Scheduler) {
+    this->Scheduler = Scheduler;
+  }
+
+  void apply(ScheduleDAGInstrs *DAG) override {
     MachineFunction &MF = DAG->MF;
     MachineRegisterInfo &MRI = MF.getRegInfo();
     const TargetRegisterInfo *TRI = MF.getSubtarget().getRegisterInfo();
@@ -513,7 +520,8 @@ class SchedWAWEdges : public WAWEdges {
     AIEPostRASchedStrategy *Scheduler =
         DAG->getBB() ? static_cast<AIEScheduleDAGMI *>(DAG)->getSchedImpl()
                      : nullptr;
-    WAWEdges::apply(DAG, Scheduler);
+    setScheduler(Scheduler);
+    WAWEdges::apply(DAG);
   }
 };
 
@@ -522,8 +530,7 @@ class SchedWAWEdges : public WAWEdges {
 // independently of the Scheduler.
 class SWPWAWEdges : public WAWEdges {
   void apply(ScheduleDAGInstrs *DAG) override {
-    AIEPostRASchedStrategy *Scheduler = nullptr;
-    WAWEdges::apply(DAG, Scheduler);
+    WAWEdges::apply(DAG);
   }
 };
 } // namespace
```